oak_functions/abi/src/lib.rs

```diff
@@ -83,4 +83,37 @@ extern "C" {
         inference_ptr_ptr: *mut *mut u8,
         inference_len_ptr: *mut usize,
     ) -> u32;
+
+    /// Reads a message, i.e., `dest_buf_len_ptr` bytes, from the channel with `channel_handle` into
```

Comment:
How about documenting the ABI in the readme file, and referencing it from here? Like the other methods in this file. Otherwise it does not really scale to document the same ABI in each potential new language (e.g. if we add Go / C++ support in the future).

Reply:
Yes, I agree the documentation should move to the ABI readme. However, I was hoping to keep it close to the definitions, while they are still changing and move it once they are stable. 

Do you think it's a problem to keep them here for now?

Reply:
SG!

Reply:
I don't think it's correct to say that it reads `dest_buf_len_ptr` bytes; first of all, that's a pointer, second, when this function is called, it does not point to anything meaningful, definitely not to the number of bytes to read. It is only used as an output parameter for the runtime to *write* out the number of bytes read. Makes sense?

Reply:
Yes, I think I reformulated this already.